src/Mvc/Mvc.Core/src/ModelBinding/Binders/SimpleTypeModelBinderProvider.cs

```diff
@@ -18,7 +18,7 @@ public class SimpleTypeModelBinderProvider : IModelBinderProvider
     {
         ArgumentNullException.ThrowIfNull(context);
 
-        if (!context.Metadata.IsComplexType)
+        if (context.Metadata.IsConvertibleType)
```

Comment:
I assume that if `IsConvertibleType` is true, then we know that the model is not correlated with a complex type.

Reply:
Exactly:

https://github.com/dotnet/aspnetcore/blob/9b84c4b982e05260077e7759f780bb0ed0be554e/src/Mvc/Mvc.Abstractions/src/ModelBinding/ModelMetadata.cs#L445